libshpool/src/daemon/server.rs

```diff
@@ -612,9 +618,10 @@ impl Server {
         conn_id: usize,
         client_stream: UnixStream,
         header: &AttachHeader,
+        user_info: &user::Info,
+        shell_env: &[(String, String)],
```

Comment:
minor nit: maybe call it `supplemental_env` or emphasize in the docstrings that these are the shpool specific shell envs, but that otherwise any other envs from the spawning shell are inherited?